File: android-app/app/src/main/java/app/calendariociclismo/android/ui/components/CategoryBadge.kt
package app.calendariociclismo.android.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import app.calendariociclismo.android.ui.theme.categoryBadgeColor
import app.calendariociclismo.android.util.LocaleHolder

/**
 * Badge de categoría UCI ("1.UWT", "WC", "1.Pro", …).
 *
 * Equivalente compose-nativo de `ios-app/.../Views/Components/CategoryBadge.swift`.
 */
@Composable
fun CategoryBadge(category: String?, modifier: Modifier = Modifier) {
    if (category.isNullOrBlank()) return
    val colors = categoryBadgeColor(category)
    Text(
        text = category.uppercase(LocaleHolder.currentState),
        style = MaterialTheme.typography.labelSmall,
        fontWeight = FontWeight.SemiBold,
        color = colors.foreground,
        modifier = modifier
            .background(
                color = colors.background,
                shape = RoundedCornerShape(3),
            )
            .padding(horizontal = 8.dp, vertical = 3.dp),
    )
}

File: ios-app/CalendarioCiclismo/Views/Components/CategoryBadge.swift
import SwiftUI

/// Badge de categoría UCI: "1.UWT", "WC", "1.Pro", etc.
struct CategoryBadge: View {
    let category: String?
    @Environment(\.accessibilityShowButtonShapes) private var showButtonShapes

    private var isHighContrast: Bool { showButtonShapes }

    var body: some View {
        if let cat = category, !cat.isEmpty {
            let colors = AppTheme.categoryBadgeColor(for: cat, highContrast: isHighContrast)
            Text(cat)
                .font(.caption2)
                .fontWeight(.semibold)
                .textCase(.uppercase)
                .padding(.horizontal, 8)
                .padding(.vertical, 3)
                .background(colors.background)
                .foregroundStyle(colors.foreground)
                .clipShape(RoundedRectangle(cornerRadius: 3))
                .overlay(
                    isHighContrast
                        ? RoundedRectangle(cornerRadius: 3).strokeBorder(colors.foreground, lineWidth: 1)
                        : nil
                )
                .accessibilityLabel(AccessibilityCategoryLabel.description(for: cat) ?? cat)
                .accessibilityIdentifier(AccessibilityID.filterButton("cat_\(cat)"))
        }
    }
}
